lib/silo/silo.rb

```diff
@@ -247,6 +239,48 @@ def push(source, dest, recursive: false)
       run_action('push.sh', env: env)
     end
 
+    def set_metadata(data)
+      if @name != data["name"] && !self.class.get_silo(name: data["name"], type: @type, creds: @creds)&.empty?
+        raise RemoteSiloExistsError, "A silo named '#{name}' already exists on remote provider '#{type.name}'"
+      end
+      File.write('/tmp/#{silo.id}_cloud_metadata.yaml', data.to_yaml)
```

Comment:
This looks like you're trying to use interpolation in single-quotes; should they be double-quotes?

Reply:
FWIW; this works, because you're using the same string each time. It's just that the string is `/tmp/#{silo.id}_cloud_metadata.yaml`, and not `/tmp/<actual-id>_cloud_metadata.yaml`. It would break if multiple instance of this command were run at once.

Reply:
I see. You're right, those should be double quotes